src/overlays/actors/ovl_En_Nb/z_en_nb.c

```diff
@@ -31,81 +114,660 @@ const ActorInit En_Nb_InitVars = {
+    s16 cutscene = this->actor.cutscene;
+    s32 i;
+
+    for (i = 0; i < arg1; i++) {
+        cutscene = ActorCutscene_GetAdditionalCutscene(cutscene);
+    }
+
+    return cutscene;
+}
+
+s32 func_80BC00AC(Actor* thisx, PlayState* play) {
+    EnNb* this = THIS;
+    s16 cutscene = func_80BC0050(this, 0);
+    s32 ret = false;
+
+    switch (this->unk_288) {
+        case 0:
+            if (!func_80BBFFD4(this, cutscene)) {
+                break;
+            }
+        // fallthrough
+        case 2:
+        case 4:
+        case 6:
+        case 8:
+            Camera_SetTargetActor(Play_GetCamera(play, ActorCutscene_GetCurrentSubCamId(cutscene)), &this->actor);
+            this->unk_288++;
+            ret = true;
+            break;
+
+        case 1:
+        case 3:
+        case 5:
+        case 7:
+            if ((this->actor.child != NULL) && (this->actor.child->update != NULL)) {
+                Camera_SetTargetActor(Play_GetCamera(play, ActorCutscene_GetCurrentSubCamId(cutscene)),
+                                      this->actor.child);
+            }
+            this->unk_288++;
+            ret = true;
+            break;
+
+        case 9:
+            ActorCutscene_Stop(cutscene);
+            this->unk_288++;
+            ret = true;
+            break;
+    }
+
+    return ret;
+}
+
+s32 func_80BC01DC(Actor* thisx, PlayState* play) {
+    EnNb* this = THIS;
+    s32 pad;
+    s32 ret = false;
+
+    switch (this->unk_288) {
+        case 0:
+            if (Player_GetMask(play) == PLAYER_MASK_ALL_NIGHT) {
+                this->unk_288 = 1;
+            } else {
+                this->unk_288 = 5;
+            }
+            break;
+
+        case 1:
+            func_8016A268(&play->state, 1, 0, 0, 0, 0);
+            this->unk_286 = 40;
+            this->unk_288 = (u16)(this->unk_288 + 1);
+            break;
+
+        case 2:
+            MREG(68) = (s16)(s32)(255.0f - (((f32)ABS_ALT(20 - this->unk_286) / 20.0f) * 255.0f));
+
+            if (this->unk_286 == 20) {
+                if (gSaveContext.eventInf[4] & 4) {
+                    play->interfaceCtx.unk_31B = 0;
+                } else {
+                    play->interfaceCtx.unk_31B = 1;
+                }
+                play->interfaceCtx.unk_31A = 6;
+                XREG(91) = 0xFF;
+            }
+
+            if (DECR(this->unk_286) == 0) {
+                this->unk_288++;
+            }
+            break;
+
+        case 3:
+            play->interfaceCtx.unk_31A = 4;
+            this->unk_288++;
+            ret = true;
+            break;
+
+        case 4:
+            play->interfaceCtx.unk_31A = 5;
+            this->unk_288++;
+            // fallthrough
+        case 5:
+            if (!(gSaveContext.eventInf[4] & 4)) {
+                gSaveContext.save.time = CLOCK_TIME(8, 0);
+                Sram_IncrementDay();
+            } else {
+                func_800FE658(120.0f);
+            }
```

Comment:
Looks like this eventInf is set or not depending on which story (although the setting's probably buried in the msgEvent script).